### PR TITLE
fix: Save As PDF from PDF Preview

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/24458.

Adds missing WebContents helpers to `ElectronExtensionsAPIClient` so that the relevant IPC events can be listened to. Also ensures that we can get the correct pdf element to print in `GetPdfElement`.

cc @deepak1556 @nornagon @zcbenz 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed an issue where `Save as PDF` from PDF Viewer Print dialog failed and sometimes crashed.

### CODE REVIEW

**Review by deepak1556 — COMMENTED** [review]
LGTM with minor changes

File: shell/renderer/printing/print_render_frame_helper_delegate.cc [hunk 1]
```diff
@@ -4,10 +4,19 @@
...
+#include "chrome/common/url_constants.h"
 #include "content/public/renderer/render_frame.h"
+#include "extensions/buildflags/buildflags.h"
 #include "third_party/blink/public/web/web_element.h"
 #include "third_party/blink/public/web/web_local_frame.h"
 
+#if BUILDFLAG(ENABLE_EXTENSIONS)
+#include "chrome/common/extensions/extension_constants.h"
+#include "extensions/common/constants.h"
+#include "extensions/renderer/guest_view/mime_handler_view/post_message_support.h"
```
[the remark this answers is not in the document]
Comment (codebytere) [reply shell/renderer/printing/print_render_frame_helper_delegate.cc:17]: oops, left this behind, thanks!

File: shell/renderer/printing/print_render_frame_helper_delegate.cc [hunk 2]
```diff
@@ -4,10 +4,19 @@
 
 #include "shell/renderer/printing/print_render_frame_helper_delegate.h"
 
+#include "chrome/common/chrome_switches.h"
+#include "chrome/common/url_constants.h"
```
[the remark this answers is not in the document]
Comment (deepak1556) [reply shell/renderer/printing/print_render_frame_helper_delegate.cc:8]: @codebytere //chrome/common headers are not removed yet ;)

**Review by MarshallOfSound — APPROVED** [review]

File: shell/renderer/printing/print_render_frame_helper_delegate.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (codebytere) [reply shell/renderer/printing/print_render_frame_helper_delegate.cc]: done! ty 🙇🏻‍♀️

**Review by deepak1556 — APPROVED** [review]

**Comment by codebytere** [comment]
Reran WOA 5 times and it flaked on the same tests every time:

> `<webview> tag loads devtools extensions registered on the parent window`

> `reporting api sends a report for a deprecation`

but they are unrelated and present on multiple other PRs

**Comment by rjlfurtado** [comment]
Thank you for your support!!!
This will certainly help many colleagues.

**Pull request merged**